### PR TITLE
feat: new reviews module

### PR DESCRIPTION
**Pull request by Leon-pix-coder**
New reviews module

### CODE REVIEW

File: src/utils/time.ts [hunk 1]
```diff
@@ -0,0 +1,24 @@
...
+    return `${days} day${days > 1 ? 's' : ''} ago`;
+  } 
+  if (hours >= 1) {
+    return `${hours} hour${hours > 1 ? 's' : ''} ago`;
+  } 
+  if (minutes >= 1) {
+    return `${minutes} minute${minutes > 1 ? 's' : ''} ago`;
+  }
+  return `just now`;
+}
```
Comment (evan-spidy) [inline src/utils/time.ts]: use dayjs to calculate

File: src/components/Textarea/index.tsx [hunk 2]
```diff
@@ -0,0 +1,86 @@
...
+  return (
+    <div className="flex flex-row items-end gap-[8px]">
+      <div className="flex flex-col py-[12px] px-[16px] bg-input rounded-[20px] flex-1">
+        <textarea
+          ref={textareaRef}
+          className={`p-0 font-questrial text-[14px] leading-[1.2] outline-none resize-none overflow-hidden appearance-none bg-input ${rootClassName}`}
+          value={text}
+          maxLength={maxLength}
+          onChange={handleChange}
+          placeholder={placeholder || "请输入..."}
```
Comment (evan-spidy) [inline src/components/Textarea/index.tsx]: do not use chinese

File: src/components/Textarea/index.tsx [hunk 3]
```diff
@@ -0,0 +1,86 @@
...
+      </div>
+      <button
+        onClick={handleSubmit}
+        className="bg-tertiary w-[40px] h-[40px] flex justify-center items-center p-[8px] rounded-[20px] shrink-0"
+        disabled={text.trim().length === 0}
+      >
+        <i
+          className={`votigram-icon-send text-[24px] ${
+            charCount === 0 ? "text-input-placeholder" : "text-primary"
+          }`}
```
Comment (evan-spidy) [inline src/components/Textarea/index.tsx]: use clsx

File: src/components/Textarea/index.tsx [hunk 4]
```diff
@@ -0,0 +1,86 @@
...
+          onChange={handleChange}
+          placeholder={placeholder || "请输入..."}
+          rows={1}
+        />
+        <span
+          className={`${
+            charCount > 0 ? "inline-block" : "hidden"
+          } mt-[10px] text-[11px] leading-[1.2] font-questrial text-input-placeholder ${
+            charCount === maxLength ? "text-danger" : ""
+          }`}
```
Comment (evan-spidy) [inline src/components/Textarea/index.tsx]: use clsx

File: src/components/List/index.tsx [hunk 5]
```diff
@@ -0,0 +1,105 @@
...
+import Item from "./components/Item";
+import clsx from "clsx";
+
+export type ListItem = {
+  id?: number;
+  title: React.ReactNode;
+  subtitle?: React.ReactNode;
+  avatar?: string;
+  content?: React.ReactNode;
+};
```
Comment (evan-spidy) [inline src/components/List/index.tsx]: create a new file components/List/type/index.ts

File: src/components/List/index.tsx [hunk 6]
```diff
@@ -0,0 +1,105 @@
...
+  return (
+    <div
+      ref={listRef}
+      role="list"
+      className={`overflow-x-hidden overflow-y-auto`}
+      style={{ height: calculatedHeight }}
+    >
+      {items.length === 0 && (
+        <div className="flex items-center justify-center h-full">
+          <span className="font-normal text-[13px] text-white font-questrial leading-[1.2]">
```
Comment (evan-spidy) [inline src/components/List/index.tsx:74]: the default font is font-questrial, so you do not need to set here

Comment (evan-spidy) [reply src/components/List/index.tsx:74]: what is leading-[1.2] ?

File: src/components/List/index.tsx [hunk 7]
```diff
@@ -0,0 +1,105 @@
...
+    // eslint-disable-next-line react-hooks/exhaustive-deps
+  }, [items, hasMore]);
+
+  const calculatedHeight = typeof height === "number" ? `${height}px` : height;
+
+  return (
+    <div
+      ref={listRef}
+      role="list"
+      className={`overflow-x-hidden overflow-y-auto`}
```
Comment (evan-spidy) [inline src/components/List/index.tsx]: do you need {``} ?

File: src/components/List/index.tsx [hunk 8]
```diff
@@ -0,0 +1,105 @@
...
+        {items.map((item, index) => (
+          <Item
+            data={item}
+            key={`${item.id}-${index}`}
+            className={itemClassname}
+          />
+        ))}
+        {items.length > 0 && !hasMore && (
+          <div className="py-[20px] flex items-center justify-center">
+            <span className="font-normal text-[13px] text-white font-questrial leading-[1.2]">
```
Comment (evan-spidy) [inline src/components/List/index.tsx:89]: same this leading-[1.2]

File: src/components/Discover/index.tsx [hunk 9]
```diff
@@ -167,15 +193,42 @@ const FileScroll = () => {
...
-      </Modal> 
+      </Modal>
+      <Drawer isVisible={false} direction="bottom">
+        <div className="relative w-full flex items-center justify-center py-[17px] px-[19px]">
+          <span className="font-outfit text-[16px] leading-[1] font-bold text-white">
+            Reviews
+          </span>
+          <i
+            className="votigram-icon-cancel text-[14px] text-white absolute top-[15px] right-[18px] cursor-pointer"
+            // onClick={() => setShowDrawer(false)}
```
Comment (evan-spidy) [inline src/components/Discover/index.tsx]: do you still need this?

File: src/utils/time.ts (hunk 1 above)
Comment (Leon-pix-coder) [reply src/utils/time.ts]: Repalced with dayjs

File: src/components/Textarea/index.tsx (hunk 4 above)
Comment (Leon-pix-coder) [reply src/components/Textarea/index.tsx]: Updated

File: src/components/List/index.tsx (hunk 5 above)
Comment (Leon-pix-coder) [reply src/components/List/index.tsx]: Updated

File: src/components/List/index.tsx (hunk 6 above)
Comment (Leon-pix-coder) [reply src/components/List/index.tsx:74]: Updated

File: src/components/List/index.tsx (hunk 8 above)
Comment (Leon-pix-coder) [reply src/components/List/index.tsx:89]: Updated

File: src/components/Discover/index.tsx (hunk 9 above)
Comment (Leon-pix-coder) [reply src/components/Discover/index.tsx]: Removed

File: src/components/Textarea/index.tsx (hunk 2 above)
Comment (Leon-pix-coder) [reply src/components/Textarea/index.tsx]: Updated

File: src/components/Textarea/index.tsx (hunk 3 above)
Comment (Leon-pix-coder) [reply src/components/Textarea/index.tsx]: Updated

File: src/components/List/index.tsx (hunk 7 above)
Comment (Leon-pix-coder) [reply src/components/List/index.tsx]: Updated

**Comment by evan-spidy** [comment]
Re-create to update PR changes

**Pull request closed without merging**